packages/medusa/src/loaders/plugins.ts

```diff
@@ -252,9 +252,11 @@ export async function registerServices(pluginDetails: PluginDetails, container:
       const loaded = require(fn).default
       const name = formatRegistrationName(fn)
 
-      if (!(loaded.prototype instanceof BaseService)) {
+      const isService = loaded.prototype instanceof BaseService
+        || loaded.prototype instanceof TransactionBaseService
```

Comment:
`TransactionBaseService` is the new base service exported from `../interfaces`

Reply:
thought, very nit: Given that `TransactionBaseService` is our new base service, would it make sense to rename it to `BaseService` abstracting away the transactional part of the concept? Will be more obvious for developers that they should default to this interface when building out their own services.

Would obv. require us to rename current `BaseService` to `LegacyBaseService` or the likes. Let me know what you think

Reply:
Got it!

I wouldn't rename the class + file of the current BaseService, but just alias the import of it such that we can use BaseService for the transactional interface.

Reply:
My previous comment was actually referring to a renaming of the actual file + class, and not so much the aliasing as you've done here. But after having thought about it, I don't think it's worth the work to rename the `TransactionBaseService` so this should be good for now 👍 @adrien2p

Reply:
👍

Reply:
Sorry - my bad, I totally see why it might have caused a bit of confusion. I meant, that we could rename the TransactionBaseService to BaseService, and then alias the import the "old" BaseService to allow for it to build, but nvm 😅

Reply:
We could definitely do that, but I have a refactoring that will probably come later that will need a BaseService and extends the TransactionBaseService, but we will be able to use either one or the other 😉